### ISSUE TITLE
False Positive Match for props validation

### ISSUE THREAD

**Issue by LeonardoGentile** [yannickcr/eslint-plugin-react#1772]
Hello, this is my setup:
**Webstorm 2018.1.1**

node: v8.11.1
npm: 5.6.0
eslint: 4.19.1
eslint-plugin-react: 7.7.0
react: 16.0.0,
react-dom: 16.0.0,
prop-types: 15.6.1

And I get false positive matches for prop-types validation:
```
class Index extends React.Component {

  constructor(props) {
    super(props);
  }

  _setActiveTab() {
    const tabStore = this.props.tabStore;  // False positive: 'tabStore is missing in props validation'
    const {id} = this.props.route.params;  // False positive: 'route.params is missing in props validation'
    tabStore.setActiveTab(id);
  }

  componentDidMount() {
    this._setActiveTab();
  }

  render() {
    const tabStore = this.props.tabStore;  // Here's is OK
    return (
      <div className={styles.whatever}> // CSS-Modules
        <h4>I am The Index Page</h4>
        <br/>
        <h4>(tabStore.tabs): </h4>
      </div>
    );
  }
}

Index.PropTypes = {
  tabStore: PropTypes.object,
  route: PropTypes.object
};
```

The problem seems to go away if I pass the props to the _setActiveTab method, that is:

```
// ...
  _setActiveTab(props) {
    const tabStore = props.tabStore;
    const {id} = props.route.params;
    tabStore.setActiveTab(id);
  }

  componentDidMount() {
    this._setActiveTab(this.props);  // passing props to method
  }

```

Is there any chance that this is coming from Webstorm and not eslint?

**Comment by LeonardoGentile** [comment]
CORRECTION: 
while typing and editing the component also this line:

```
render() {
    const tabStore = this.props.tabStore;  // <== THIS LINE
    return (
      <div className={styles.whatever}> // CSS-Modules
        <h4>I am The Index Page</h4>
        <br/>
        <h4>(tabStore.tabs): </h4>
      </div>
    );
  }
```
will randomly trigger a false match.
Simple example: if I remove the className attribute to the div

```
// from this
<div className={styles.whatever}> // CSS-Modules
// to this:
<div > 
```
Then the prop-type validation triggers a false match: 'tabStore is missing in props validation'

**Comment by ljharb** [comment]
@LeonardoGentile it’s not a false positive, the property name is supposed to be `propTypes`, not `PropTypes`. There’s a separate “no typos” rule you may wish to look into.
